indexer/feature_data.cpp

```diff
@@ -213,29 +217,59 @@ bool FeatureParams::AddHouseNumber(string const & ss)
   return true;
 }
 
-void FeatureParams::AddStreetAddress(string const & s)
+void FeatureParams::AddStreet(string s)
 {
-  m_street = s;
-
   // Erase bad chars (\n) because we write addresses to txt file.
-  m_street.erase(remove_if(m_street.begin(), m_street.end(), IsBadChar()), m_street.end());
+  s.erase(remove_if(s.begin(), s.end(), IsBadChar()), s.end());
 
   // Osm likes to put house numbers into addr:street field.
-  size_t i = m_street.find_last_of("\t ");
+  size_t i = s.find_last_of("\t ");
   if (i != string::npos)
   {
-    ++i;
     uint64_t n;
-    if (strings::to_uint64(m_street.substr(i), n))
-      m_street.erase(i);
+    if (strings::to_uint64(s.substr(i+1), n))
```

Comment:
Probably we need to use a smarter version of the house numbers classifier here. Please, add a TODO (@ygorshenin) comment here.